### ISSUE TITLE
Logstash parse issues

### ISSUE THREAD

**Issue by ARPLink** [elastic/elasticsearch#14131]
Hello,

We have the following usercase. I have a log containing fields enclosed in "" and separated by ;

The grok filter is set like this:

"%{DATA:1}";"%{DATA:2}";"%{DATA:3}";"%{DATA:4}";"%{DATA:5}";"%{DATA:6}";"%{DATA:7}";"%{DATA:8}";"%{DATA:9}";"%{DATA:10}";"%{DATA:11}";"%{DATA:12}";"%{DATA:13}";"%{DATA:14}";"%{DATA:15}";"%{DATA:16}";"%{DATA:17}";"%{DATA:18}";"%{DATA:19}";"%{DATA:20}";"%{DATA:21}";"%{DATA:22}";"%{DATA:23}";"%{DATA:24}";"%{DATA:25}";"%{DATA:26}";"%{DATA:27}";"%{DATA:28}";"%{DATA:29}";"%{DATA:30}";"%{DATA:31}";"%{DATA:32}";"%{DATA:33}";"%{DATA:34}";"%{DATA:35}";"%{DATA:36}";"%{DATA:37}";"%{DATA:38}";"%{DATA:39}";"%{DATA:40}";"%{DATA:41}";"%{DATA:42}";"%{DATA:43}";"%{DATA:44}";"%{DATA:45}";"%{DATA:46}";"%{DATA:47}";"%{TIMESTAMP_ISO8601:48}";"%{DATA:49}";"%{DATA:50}";"%{DATA:51}";"%{DATA:52}";"%{DATA:53}";"%{DATA:54}";"%{DATA:55}";"%{DATA:56}";"%{DATA:57}"

Everything works perfectly except logstash spits out an error for "%{DATA:28}". I tried DATA, QS, NOTSPACE you name it...

This field contains multiple formats of data, so I need logstash to parse all of them. Somethimes field 28 shows "1", sometimes it looks similar to "AAAAA-A0000A0A-1111-2222-A3AA-A444A4444AA4" and other times it's a number sequence similar to "000-1111-2222-33333". I checked the logs and this is what I get(see below). Basically I tell it to parse some bulk data and it ignores me and tries to parse it how it sees fit. At the bottom of the page is an extras from GrokDebugger and it sheds a little light in my issue.

org.elasticsearch.index.mapper.MapperParsingException: failed to parse [28]
        at org.elasticsearch.index.mapper.core.AbstractFieldMapper.parse(AbstractFieldMapper.java:411)
        at org.elasticsearch.index.mapper.object.ObjectMapper.serializeValue(ObjectMapper.java:706)
        at org.elasticsearch.index.mapper.object.ObjectMapper.parse(ObjectMapper.java:497)
        at org.elasticsearch.index.mapper.DocumentMapper.parse(DocumentMapper.java:544)
        at org.elasticsearch.index.mapper.DocumentMapper.parse(DocumentMapper.java:493)
        at org.elasticsearch.index.shard.IndexShard.prepareCreate(IndexShard.java:466)
        at org.elasticsearch.action.bulk.TransportShardBulkAction.shardIndexOperation(TransportShardBulkAction.java:418)
        at org.elasticsearch.action.bulk.TransportShardBulkAction.shardOperationOnPrimary(TransportShardBulkAction.java:148)
        at org.elasticsearch.action.support.replication.TransportShardReplicationOperationAction$PrimaryPhase.performOnPrimary(TransportShardReplicationOperationAction.java:574)
        at org.elasticsearch.action.support.replication.TransportShardReplicationOperationAction$PrimaryPhase$1.doRun(TransportShardReplicationOperationAction.java:440)
        at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:36)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
        at java.lang.Thread.run(Thread.java:745)
Caused by: org.elasticsearch.index.mapper.MapperParsingException: failed to parse date field [1383-5494-1131-58264], tried both date format [dateOptionalTime], and timestamp number with locale []
        at org.elasticsearch.index.mapper.core.DateFieldMapper.parseStringValue(DateFieldMapper.java:617)
        at org.elasticsearch.index.mapper.core.DateFieldMapper.innerParseCreateField(DateFieldMapper.java:535)
        at org.elasticsearch.index.mapper.core.NumberFieldMapper.parseCreateField(NumberFieldMapper.java:239)
        at org.elasticsearch.index.mapper.core.AbstractFieldMapper.parse(AbstractFieldMapper.java:401)
        ... 13 more
Caused by: java.lang.IllegalArgumentException: Invalid format: "1383-5494-1131-58264" is malformed at "4-1131-58264"
        at org.elasticsearch.common.joda.time.format.DateTimeParserBucket.doParseMillis(DateTimeParserBucket.java:187)
        at org.elasticsearch.common.joda.time.format.DateTimeFormatter.parseMillis(DateTimeFormatter.java:780)
        at org.elasticsearch.index.mapper.core.DateFieldMapper.parseStringValue(DateFieldMapper.java:612)
        ... 16 more

GROK DEBUGGER:
 Basically, if I let logstash interpret my field as he consideres fit, it will do something like this. The problem is that even when I define my field as DATA, it treats it as below:

"AAAAA%{BASE16FLOAT}-1111%{ISO8601_TIMEZONE}1-A3AA-A444A4444AA4"

**Comment by clintongormley** [comment]
Hi @ARPLink 

I suggest you ask this question in the forums: https://discuss.elastic.co/
